src/common/bsod.cpp

```diff
@@ -301,7 +302,32 @@ void draw_error_screen(const uint16_t error_code_short) {
         /// draw short URL
         error_url_short(qr_text, sizeof(qr_text), error_code);
         // this MakeRAM is safe - qr_text is a local buffer on stack
-        render_text_align(Rect16(0, 293, display::GetW(), display::GetH() - 293), string_view_utf8::MakeRAM((const uint8_t *)qr_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        render_text_align(Rect16(0, 255, display::GetW(), display::GetH() - 255), string_view_utf8::MakeRAM((const uint8_t *)qr_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+
+        /// draw footer information
+        /// fw version, hash, [apendix], [fw signed]
+        /// fw version
+        char fw_version[32];
+        snprintf(fw_version, sizeof(fw_version), "%s%s", project_version, project_version_suffix_short);
+        render_text_align(Rect16(6, 290, 80, 10), string_view_utf8::MakeRAM((const uint8_t *)fw_version), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        /// hash
+        if (!qr_privacy) {
+            char p_code[] = {};
+            printerCode(p_code);
+            render_text_align(Rect16(98, 290, 64, 10), string_view_utf8::MakeRAM((const uint8_t *)p_code), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        }
+        /// [apendix, fw signed]
+        /// TODO: fw signed is not available ATM
+        /// signed fw
+        if (0) {
+            char signed_fw_str[4] = "[S]";
```

Comment:
static const, it doesn't have to be in RAM ... and use MakeCPUFLASH then (even though it is the same for now)